db/external_sst_file_ingestion_job.cc

```diff
@@ -393,18 +412,48 @@ Status ExternalSstFileIngestionJob::Run() {
   }
   // It is safe to use this instead of LastAllocatedSequence since we are
   // the only active writer, and hence they are equal
-  SequenceNumber last_seqno = versions_->LastSequence();
+  SequenceNumber batch_start_last_seqno = versions_->LastSequence();
   edit_.SetColumnFamily(cfd_->GetID());
   // The levels that the files will be ingested into
 
-  for (IngestedFileInfo& f : files_to_ingest_) {
+  std::optional<int> prev_batch_uppermost_level;
+  for (auto& batch : file_batches_to_ingest_) {
+    SequenceNumber batch_end_last_seqno = 0;
+    int batch_uppermost_level = 0;
+    status = AssignLevelsForOneBatch(
+        batch, super_version, force_global_seqno, batch_start_last_seqno,
+        &batch_end_last_seqno, &batch_uppermost_level,
+        prev_batch_uppermost_level);
+    if (!status.ok()) {
+      return status;
+    }
+    batch_start_last_seqno = batch_end_last_seqno;
```

Comment:
Should take max?

Reply:
I see. I think with your change, we should only need each batch to have increasing sequence number. But this should be fine too.

Reply:
Yeah, I think keeping it like this is simpler, and it's a matter of using a few more sequence numbers, which should be fine.